tests/rbd/test_rbd_compression.py

```diff
@@ -0,0 +1,133 @@
+"""Test case covered -
+    CEPH-83574644 - Validate "rbd_compression_hint" config
+    settings on globally, Pool level, and image level.
+
+    Pre-requisites :
+    1. Cluster must be up and running with capacity to create pool
+       (At least with 64 pgs)
+    2. We need atleast one client node with ceph-common package,
+       conf and keyring files
+
+    Test Case Flow:
+    1. Create a pool and an Image, write some data on it
+    2. set bluestore_compression_mode to passive to enable rbd_compression_hint feature
+    3. verify "rbd_compression_hint" to "compressible" on global, pool and image level
+    4. verify "rbd_compression_hint" to "incompressible" on global, pool and image level
+    5. Repeat the above steps for ecpool
+"""
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_compression(rbd, pool_type, **kw):
+    """
+    Run rbd_compression_hint test on supported options,
+    i.e compressible and incompressible.
+
+    Args:
+        rbd: RBD object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        **kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 for failure.
+    """
+    config = kw.get("config")
+
+    pool_name = config[pool_type]["pool"]
+    image_name = config[pool_type]["image"]
+    image_spec = pool_name + "/" + image_name
+
+    entity = {"global": "global", "pool": pool_name, "image": image_spec}
+
+    try:
+        # set bluestore_compression_mode to passive
+        rbd.set_ceph_config("osd", "bluestore_compression_mode", "passive")
+
+        if (
+            rbd.get_ceph_config(
+                "osd", "bluestore_compression_mode", output=True
+            ).strip()
+            != "passive"
+        ):
+            log.error("Not able to set bluestore_compression_mode to passive")
+            return 1
+
+        log.info("Able to set bluestore_compression_mode to passive")
+
+        for option in ["compressible", "incompressible"]:
```

Comment:
[optional] we are verifying set and get twice on global level, we can think to run on global only once.

Reply:
I don't think so, looks like you got confused with `entity` global level has entity of type global.